app/controllers/app/training_controller.rb

```diff
@@ -8,11 +9,13 @@ def new
   def create
     exercise = Exercise.find(params[:training][:exercise_id])
     @training = exercise.trainings.build(training_params)
+    @training.exercise_time = Time.now
+    current_user.trainings << @training
```

Comment:
Maybe you don't need an Exercise object here, right?
The reference to an exercise is enough to create a `Training`.
And I can see `exercise_id` is already in `training_params`.

So probably you can do something like this should work:

```ruby
current_user.trainings.build(training_params)
```

I don't remember the meaning of `exercise_time`, but maybe it is a good idea if you move it to `training_params` method too, like this:

```ruby
def training_params
  params
    .require(:training)
    .permit(:weight, :number_of_sets, :number_of_reps, :exercise_id)
    .merge(exercise_time: Time.current)
end
```

Check out this article about timezones and you will understand why I changed from `Time.now` to `Time.current`: https://thoughtbot.com/blog/its-about-time-zones#a-summary-of-do39s-and-don39ts-with-time-zones